src/test/java/org/broadinstitute/hellbender/tools/spark/pipelines/metrics/QualityScoreDistributionSparkIntegrationTest.java

```diff
@@ -0,0 +1,183 @@
+package org.broadinstitute.hellbender.tools.spark.pipelines.metrics;
+
+import htsjdk.samtools.SAMTag;
+import htsjdk.samtools.SAMUtils;
+import org.broadinstitute.hellbender.CommandLineProgramTest;
+import org.broadinstitute.hellbender.cmdline.StandardArgumentDefinitions;
+import org.broadinstitute.hellbender.tools.spark.pipelines.metrics.QualityScoreDistributionSpark.Counts;
+import org.broadinstitute.hellbender.utils.read.ArtificialReadUtils;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+import org.broadinstitute.hellbender.utils.test.ArgumentsBuilder;
+import org.broadinstitute.hellbender.utils.test.BaseTest;
+import org.broadinstitute.hellbender.utils.test.IntegrationTestSpec;
+import org.junit.Assert;
+import org.testng.annotations.Test;
+
+import java.io.File;
+import java.io.IOException;
+
+public class QualityScoreDistributionSparkIntegrationTest  extends CommandLineProgramTest {
+
+    //NOTE: these tests use the same data and results as the non-spark ones, by design
+
+    private static final File TEST_DATA_DIR = new File(getTestDataDir(), "picard/analysis/QualityScoreDistribution");
+
+    @Override
+    public String getTestedClassName() {
+        return QualityScoreDistributionSpark.class.getSimpleName();
+    }
+
+    @Test
+    public void testAccumulators() throws Exception {
+        final long[] qs= new long[128];
+        final long[] oqs= new long[128];
+
+        final Counts counts = new Counts(false);
+        final long[] initQs = counts.getQualCounts();
+        final long[] initOQs = counts.getOrigQualCounts();
+        Assert.assertArrayEquals(initQs, qs);
+        Assert.assertArrayEquals(initOQs, oqs);
+
+        final GATKRead read1 = ArtificialReadUtils.createArtificialRead("aa".getBytes(), new byte[]{50, 50}, "2M");
+        read1.setAttribute(SAMTag.OQ.name(), SAMUtils.phredToFastq(new byte[]{30, 40}));
+        counts.addRead(read1);
+
+        qs[50]+=2;//length of the read
```

Comment:
I think a better comment would be `there are two bases in the read with a quality score of 50`.